From 08d972d38096a0deff08bb1651c32450f08cde0d Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Fri, 3 Aug 2018 16:04:48 +0800
Subject: [PATCH 14/40] Fix issue that the position of QuickSetting is improper
 on DUT (IVI)

NotificationPanelView uses the width of notification stack Scroller to
estimate if notification stack Scroller and QuickSetting to be applied
the offset. That may cause the position of QuickSetting is improper if
the width of notification stack Scroller is smaller than the width of
QuickSetting.

The solution is that comparing the width of notification stack Scroller
with the width of QuickSetting and using the larger one to estimate if
notification stack Scroller and QuickSetting to be applied the offset.

Test:
1. Click on the top right of Status Bar.
2. Show notifications

Tracked-On: OAM-67772

Change-Id: I5039d0ae24f93ca4d9e3b412ea257773372adad9
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 .../systemui/statusbar/phone/NotificationPanelView.java     | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java
index ed4c7b7f474f..b879d62183c8 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NotificationPanelView.java
@@ -2560,7 +2560,11 @@ public class NotificationPanelView extends PanelView implements
      * @param x the x-coordinate the touch event
      */
     protected void updateVerticalPanelPosition(float x) {
-        if (mNotificationStackScroller.getWidth() * 1.75f > getWidth()) {
+        int width = mQsFrame.getWidth();
+        if (width < mNotificationStackScroller.getWidth()) {
+            width = mNotificationStackScroller.getWidth();
+        }
+        if (width * 1.75f > getWidth()) {
             resetVerticalPanelPosition();
             return;
         }
-- 
2.17.1

